core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -269,6 +269,24 @@ private[spark] object Utils extends Logging {
     file.setExecutable(true, true)
   }
 
+  /**
+   * Move data to trash on truncate table given
+   * spark.sql.truncate.trash.interval is positive
+   */
+  def moveToTrashIfEnabled(
+      fs: FileSystem,
+      partitionPath: Path,
+      trashInterval: Int,
+      hadoopConf: Configuration): Unit = {
+    if (trashInterval < 0) {
+      fs.delete(partitionPath, true)
+    } else {
+      logDebug(s"will move data ${partitionPath.toString} to trash")
+      hadoopConf.setInt("fs.trash.interval", trashInterval)
```

Comment:
note this is not always honored: if server side has this configured then the client side one will be ignored

Reply:
Yes thanks for mentioning, I'll add that in configuration description.